neo4j/internal/bolt/hydrator.go

```diff
@@ -709,25 +717,71 @@ func (h *hydrator) point3d(n uint32) interface{} {
 
 func (h *hydrator) dateTimeOffset(n uint32) interface{} {
 	h.unp.Next()
-	secs := h.unp.Int()
+	seconds := h.unp.Int()
 	h.unp.Next()
-	nans := h.unp.Int()
+	nanos := h.unp.Int()
 	h.unp.Next()
-	offs := h.unp.Int()
-	t := time.Unix(secs, nans).UTC()
-	l := time.FixedZone("Offset", int(offs))
-	return time.Date(t.Year(), t.Month(), t.Day(), t.Hour(), t.Minute(), t.Second(), t.Nanosecond(), l)
+	offset := h.unp.Int()
+	// time.Time in local timezone, e.g. 15th of June 2020, 15:30 in Paris (UTC+2h)
+	unixTime := time.Unix(seconds, nanos)
+	// time.Time computed in UTC timezone, e.g. 15th of June 2020, 13:30 in UTC
+	utcTime := unixTime.UTC()
+	// time.Time **copied** as-is in the target timezone, e.g. 15th of June 2020, 13:30 in target tz
+	timeZone := time.FixedZone("Offset", int(offset))
+	return time.Date(
+		utcTime.Year(),
+		utcTime.Month(),
+		utcTime.Day(),
+		utcTime.Hour(),
+		utcTime.Minute(),
+		utcTime.Second(),
+		utcTime.Nanosecond(),
+		timeZone,
+	)
+}
+
+func (h *hydrator) utcDateTimeOffset(n uint32) interface{} {
+	h.unp.Next()
+	seconds := h.unp.Int()
+	h.unp.Next()
+	nanos := h.unp.Int()
+	h.unp.Next()
+	offset := h.unp.Int()
+	timeZone := time.FixedZone("Offset", int(offset))
+	return time.Unix(seconds, nanos).In(timeZone)
 }
 
 func (h *hydrator) dateTimeNamedZone(n uint32) interface{} {
+	h.unp.Next()
+	seconds := h.unp.Int()
+	h.unp.Next()
+	nanos := h.unp.Int()
+	h.unp.Next()
+	zone := h.unp.String() // time.Time in local timezone, e.g. 15th of June 2020, 15:30 in Paris (UTC+2h)
+	unixTime := time.Unix(seconds, nanos)
+	// time.Time computed in UTC timezone, e.g. 15th of June 2020, 13:30 in UTC
+	utcTime := unixTime.UTC()
+	// time.Time **copied** as-is in the target timezone, e.g. 15th of June 2020, 13:30 in target tz
+	l, err := time.LoadLocation(zone)
+	if err != nil {
+		h.setErr(&db.ProtocolError{
+			MessageType: "dateTimeNamedZone",
+			Field:       "location",
+			Err:         err.Error(),
+		})
+		return nil
+	}
+	return time.Date(utcTime.Year(), utcTime.Month(), utcTime.Day(), utcTime.Hour(), utcTime.Minute(), utcTime.Second(), utcTime.Nanosecond(), l)
```

Comment:
Minor layout niggle: Differing format to the return in dateTimeOffset which puts each element on a new line.